File: android/app/src/main/java/com/baatcheet/app/domain/usecase/auth/IsAuthenticatedUseCase.kt
package com.baatcheet.app.domain.usecase.auth

import com.baatcheet.app.domain.repository.AuthRepository
import javax.inject.Inject

/**
 * Is Authenticated Use Case
 * 
 * Checks if user is currently authenticated.
 */
class IsAuthenticatedUseCase @Inject constructor(
    private val authRepository: AuthRepository
) {
    /**
     * Check if user is authenticated
     */
    operator fun invoke(): Boolean {
        return authRepository.isAuthenticated()
    }
}

File: ios/iOSBaseProject/Domain/UseCases/Auth/IsAuthenticatedUseCase.swift
//
//  IsAuthenticatedUseCase.swift
//  BaatCheet
//
//  Domain layer - Check Authentication Status Use Case
//

import Foundation

/// Use case for checking if user is authenticated.
protocol IsAuthenticatedUseCaseProtocol {
    func execute() -> Bool
}

final class IsAuthenticatedUseCase: IsAuthenticatedUseCaseProtocol {
    private let repository: AuthRepositoryProtocol
    
    init(repository: AuthRepositoryProtocol) {
        self.repository = repository
    }
    
    func execute() -> Bool {
        return repository.isAuthenticated()
    }
}
